components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsPolyglotGraphBuilderFactory.java

```diff
@@ -44,50 +44,55 @@
     private static final Log LOG = LogFactory.getLog(JsPolyglotGraphBuilderFactory.class);
     private static final String JS_BINDING_CURRENT_CONTEXT = "JS_BINDING_CURRENT_CONTEXT";
 
-    private ScriptEngine engine;
+    private GraalJSEngineFactory factory;
 
     public void init() {
-
+        factory = new GraalJSEngineFactory();
     }
 
-    public static void restoreCurrentContext(AuthenticationContext context, ScriptEngine engine)
-        throws FrameworkException {
+    public static void restoreCurrentContext(AuthenticationContext authContext, Context context)
+            throws FrameworkException, IOException {
```

Comment:
from where we are getting this `IOException` better to catch it and throw as FrameworkException

Reply:
That was unnecessary. Removed the IOException.